_posts/2021-01-13-hapi-fhir.md

```diff
@@ -307,8 +306,13 @@ public class PatientProvider implements IResourceProvider {
 }
 
 ```
+In the previous code snippet I want to highlight some important details like: 
+- The MethodOutcome object must be returned on update and create methods. This object contains the identity of the created resource. 
+On a delete and validate method you have a choice between void and MethodOutcome.
+- Operation outcomes are sets of error, warning and information messages that provide detailed information about the outcome of an attempted system operation. 
+They are provided as a direct system response, or component of one, and provide information about the outcome of the operation.
```

Comment:
I dont fully understand what you mean with this sentence: "They are provided as a direct system response, or component of one, and provide information about the outcome of the operation."

Reply:
I meant that the operationOutcome can be used as a direct (or component) response from the server. For example when the method fails the Operation Outcome can provide more information about the outcome. This can be used to provide meaningful error messages.